src/UserDeveloperConverter.php

```diff
@@ -104,7 +104,11 @@ public function convertUser(UserInterface $user): UserToDeveloperConversionResul
       $setter = 'set' . ucfirst($developer_prop);
       $getter = 'get' . ucfirst($developer_prop);
       if ($user->get($base_field)->value !== $developer->{$getter}()) {
-        $developer->{$setter}($user->get($base_field)->value);
+        $value = $user->get($base_field)->value;
+        // Make sure we don't set first name and last name when nothing to set.
+        if (!in_array($base_field, ['first_name', 'last_name']) && empty($value)) {
```

Comment:
This is not going to work. As I described in https://github.com/apigee/apigee-edge-drupal/pull/277#issuecomment-539979633, you cannot create a developer in Apigee Edge without a first name or a last name.

Install a clean Drupal site. Create n+1 users, install the Apigee Edge module and configure the connection. Try to run the developer sync on the UI, it will fail with errors like this:

![image](https://user-images.githubusercontent.com/1755573/67860875-53cccf00-fb1f-11e9-8033-9ac21d136275.png)

![image](https://user-images.githubusercontent.com/1755573/67860857-47e10d00-fb1f-11e9-96d6-c9db7a68497f.png)

(users created earlier than someone enabled the Apigee Edge module does not have a first name and last name that the system could send to Apigee Edge when it tries to create/update developers.)